Multisig/Logic/Models/Safe.swift

```diff
@@ -206,20 +206,18 @@ extension Safe: Identifiable {
 
 extension Safe {
     func update(from info: SafeInfoRequest.ResponseType) {
-        DispatchQueue.main.async {
-            self.contractVersion = info.version
-            App.shared.coreDataStack.saveContext()
-        }
-
         threshold = info.threshold.value
         ownersInfo = info.owners.map { $0.addressInfo }
         implementationInfo = info.implementation.addressInfo
         nonce = info.nonce.value
         modulesInfo = info.modules?.map { $0.addressInfo }
         fallbackHandlerInfo = info.fallbackHandler?.addressInfo
         guardInfo = info.guard?.addressInfo
-                            
-        NotificationCenter.default.post(name: .selectedSafeUpdated, object: self)
+        DispatchQueue.main.async {
+            self.contractVersion = info.version
+            App.shared.coreDataStack.saveContext()
+            NotificationCenter.default.post(name: .selectedSafeUpdated, object: self)
```

Comment:
@elgatovital this line leads to an inifinit loop on the safe settings screen. What was the reason to add it?

Reply:
fixed

Reply:
header with send/receive buttons needs safe info to decide where to navigate when user clicks send button: add owner key screen or send asset screen, depending whether safe is read only. This propagates safe info to the view controller of the header.